control-plane-operator/controllers/hostedcontrolplane/csi/kubevirt/kubevirt.go

```diff
@@ -122,12 +123,46 @@ func getContentsOrDie(file string) []byte {
 	return b
 }
 
-func reconcileInfraConfigMap(cm *corev1.ConfigMap, infraID string) error {
-	storageClassEnforcement := "allowDefault: true\nallowAll: false\n"
+func getStorageDriverType(hcp *hyperv1.HostedControlPlane) hyperv1.KubevirtStorageDriverConfigType {
+
+	storageDriverType := hyperv1.DefaultKubevirtStorageDriverConfigType
+
+	if hcp.Spec.Platform.Kubevirt != nil &&
+		hcp.Spec.Platform.Kubevirt.StorageDriver != nil &&
+		hcp.Spec.Platform.Kubevirt.StorageDriver.Type != "" {
+
+		storageDriverType = hcp.Spec.Platform.Kubevirt.StorageDriver.Type
+	}
+	return storageDriverType
+}
+
+func reconcileInfraConfigMap(cm *corev1.ConfigMap, hcp *hyperv1.HostedControlPlane) error {
+	var storageClassEnforcement string
+
+	storageDriverType := getStorageDriverType(hcp)
+
+	switch storageDriverType {
+	case hyperv1.ManualKubevirtStorageDriverConfigType:
+		allowedSC := []string{}
+
+		if hcp.Spec.Platform.Kubevirt.StorageDriver.Manual != nil {
+			for _, mapping := range hcp.Spec.Platform.Kubevirt.StorageDriver.Manual.StorageClassMapping {
+				allowedSC = append(allowedSC, mapping.InfraStorageClassName)
+			}
+		}
+
+		storageClassEnforcement = fmt.Sprintf("allowAll: false\nallowList: [%s]\n", strings.Join(allowedSC, ", "))
+	case hyperv1.NoneKubevirtStorageDriverConfigType:
+		storageClassEnforcement = "allowDefault: false\nallowAll: false\n"
+	case hyperv1.DefaultKubevirtStorageDriverConfigType:
+		storageClassEnforcement = "allowDefault: true\nallowAll: false\n"
+	default:
+		storageClassEnforcement = "allowDefault: true\nallowAll: false\n"
+	}
 
 	cm.Data = map[string]string{
 		"infraClusterNamespace":        cm.Namespace,
-		"infraClusterLabels":           fmt.Sprintf("%s=%s", hyperv1.InfraIDLabel, infraID),
+		"infraClusterLabels":           fmt.Sprintf("%s=%s", hyperv1.InfraIDLabel, hcp.Spec.InfraID),
 		"infraStorageClassEnforcement": storageClassEnforcement,
```

Comment:
Correct me if I'm wrong, but it appears that the value of `storageClassEnforcement` becomes the value of the `INFRA_STORAGE_CLASS_ENFORCEMENT` environment variable for the CSI driver via the `driver-config` config map?

So changing the type/mapping will update the configmap.  But that will have no effect on the driver once it is running.  Is that "good enough?"  Should the driver be changed read this configmap key as a file instead?